main.go

```diff
@@ -61,6 +61,7 @@ type HTTPProbe struct {
 	TLSConfig              config.TLSConfig  `yaml:"tls_config"`
 	Protocol               string            `yaml:"protocol"`              // Defaults to "tcp".
 	PreferredIpProtocol    string            `yaml:"preferred_ip_protocol"` // Defaults to "ip6".
+	DisableGzipEncoding    bool              `yaml:"disable_gzip_encoding"`
```

Comment:
Agee about the naming. Initially I named it `AcceptGzipCompression`, which feels better, but it was a bit troublesome to figure out how to handle the default value (ie, distinguish between the user setting `AcceptGzipCompression: false` and not setting anything at all).
I don't think we should disable it for everyone, though. At least Akamai, and possibly others, has some weird behavior where they drop headers if the origin server sends compressed and the client requests non-compressed. Being clear about what is going on and allowing it to be toggled seems much better.